### PR TITLE
Feat/getStorageAt

### PR DESCRIPTION
**Pull request by trangnv**
Time spent on this PR: 5h

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: N/A

# What is the new behavior?

- implementing `eth_getStorageAt`
-
-

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

# Other information

### CODE REVIEW

File: core/src/lightclient/mod.rs [hunk 1]
```diff
@@ -278,80 +268,56 @@ impl StarknetClient for StarknetClientImpl {
...
+        let bytes = index.to_be_bytes();
 
-                let status_info = SyncInfo {
-                    starting_block,
-                    current_block,
-                    highest_block,
-                    warp_chunks_amount,
-                    warp_chunks_processed,
-                };
+        let f = FieldElement::from_bytes_be(&bytes).unwrap();
```
Comment (Eikix) [inline core/src/lightclient/mod.rs:304]: Let's switch from using `.unwrap()` to handling if `from_bytes_be` fails, you will find other examples in the codebase where we handle from_bytes or other FieldElement methods that return a result/option using various methods (ok_or, unwrap_or_else, etc.)

File: rpc/src/eth_rpc.rs [hunk 2]
```diff
@@ -393,8 +380,13 @@ impl EthApiServer for KakarotEthRpc {
         _address: Address,
         _index: U256,
         _block_number: Option<BlockId>,
-    ) -> Result<H256> {
-        todo!()
+    ) -> Result<Bytes> {
```
Comment (Eikix) [inline rpc/src/eth_rpc.rs:383]: Nice catch! It seems reth's types are wrong. 
https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-api/src/eth.rs#L115

You can open an issue on their repo if you'd like to signal they're not respecting the ethereum json-rpc API https://ethereum.github.io/execution-apis/api-documentation/

File: rpc/src/eth_rpc.rs [hunk 3]
```diff
@@ -393,8 +380,13 @@ impl EthApiServer for KakarotEthRpc {
...
+        let starknet_block_id = ethers_block_id_to_starknet_block_id(_block_number.unwrap())?;
```
Comment (Eikix) [inline rpc/src/eth_rpc.rs:384]: Let's not unwrap here but rather handle the error

File: rpc/src/eth_rpc.rs [hunk 4]
```diff
@@ -486,15 +478,15 @@ impl EthApiServer for KakarotEthRpc {
     }
 
     async fn is_mining(&self) -> Result<bool> {
-        Err(jsonrpsee::core::Error::Custom("Unsupported method: eth_mining. See available methods at https://github.com/sayajin-labs/kakarot-rpc/blob/main/docs/rpc_api_status.md".to_string()))
+        todo!()
```
Comment (Eikix) [inline rpc/src/eth_rpc.rs:481]: You should rebase on main:)!

File: core/src/lightclient/mod.rs [hunk 5]
```diff
@@ -278,80 +268,56 @@ impl StarknetClient for StarknetClientImpl {
...
+    async fn storage_at(
+        &self,
+        ethereum_address: Address,
+        index: U256,
+        starknet_block_id: StarknetBlockId,
+    ) -> Result<Bytes, LightClientError> {
+        // Convert the Ethereum address to a hex-encoded string
+        let address_hex = hex::encode(ethereum_address);
+        // Convert the hex-encoded string to a FieldElement
+        let ethereum_address_felt = FieldElement::from_hex_be(&address_hex).map_err(|e| {
```
Comment (Eikix) [inline core/src/lightclient/mod.rs:280]: nice error handling! let's reproduce it and replace .unwrap with this structure

**Review by Eikix — CHANGES_REQUESTED** [review]
Nice PR! Needs some reasonable amount of work to get it to a mergeable state:)

**Comment by trangnv** [comment]
will reopen soon, been sick with covid recently

**Comment by Eikix** [comment]
I'm sorry to hear that, I hope you get well soon and rest well!

**Pull request closed without merging**